packages/react-positioning/src/usePopper.ts

```diff
@@ -388,14 +388,14 @@ export function usePopper(
       popperInstanceRef.current?.destroy();
       popperInstanceRef.current = null;
     };
-  }, [options.enabled, options.target]);
+  }, [handlePopperUpdate, options.enabled, options.target]);
   useIsomorphicLayoutEffect(() => {
     if (!isFirstMount) {
       popperInstanceRef.current?.setOptions(
         resolvePopperOptions(options.target || targetRef.current, containerRef.current, arrowRef.current),
       );
     }
-  }, [resolvePopperOptions]);
+  }, [arrowRef, containerRef, isFirstMount, options.target, resolvePopperOptions, targetRef]);
```

Comment:
@ling1726 can you confirm that it's correct to add these deps of `isFirstMount` and `options.target`, which seem to have been mistakenly missing from the deps array?

Reply:
`isFirstMount` should never change after mount, you can put it there if you want

The effect before this one will create a new popper instance if target changes and resolve options immediately, take a look at `handlePopperUpdate`. The two effects are mutually exclusive, and this effect in question, only runs if the options need to be resolved without creating a new instance. It could potentially be more expensive to run this effect too on `target`, but the functionality should be the same

Reply:
I would propose to make the exception and add what I just wrote as a comment

Reply:
@ling1726 is right, this effect should run only when a reference for `resolvePopperOptions` changes.

@behowell please add an exception there.